hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java

```diff
@@ -712,4 +847,421 @@ static Throwable peel(Throwable e) {
+            }
+
+            if (isEofOnUnclosedQuote() && quoteStart >= 0) {
+                throw new EOFError(-1, quoteStart, "Missing closing quote",
+                        line.charAt(quoteStart) == '\'' ? "quote" : "dquote");
+            }
+
+            if (oneLineCommentStart != -1) {
+                throw new EOFError(-1, cursor, "One line comment");
+            }
+
+            if (multiLineCommentStart != -1) {
+                throw new EOFError(-1, cursor, "Missing end of comment", "**");
+            }
+
+            if (containsNonWhitespaceData &&
+                    (lastSemicolonIdx == -1 || lastSemicolonIdx >= cursor)) {
+                throw new EOFError(-1, cursor, "Missing semicolon (;)");
+            }
+        }
+
+    }
+
+    private void executeSqlCmd(
+            JetInstance jet,
+            String command,
+            Terminal terminal,
+            AtomicReference<SqlResult> activeSqlResult
+    ) {
+        PrintWriter out = terminal.writer();
+        try (SqlResult sqlResult = jet.getSql().execute(command)) {
+            activeSqlResult.set(sqlResult);
+
+            // if it's a result with an update count, just print it
+            if (sqlResult.updateCount() != -1) {
+                String message = new AttributedStringBuilder()
+                        .style(AttributedStyle.BOLD.foreground(PRIMARY_COLOR))
+                        .append("OK")
+                        .toAnsi();
+                out.println(message);
+                return;
+            }
+            SqlRowMetadata rowMetadata = sqlResult.getRowMetadata();
+            int[] colWidths = determineColumnWidths(rowMetadata);
+            Alignment[] alignments = determineAlignments(rowMetadata);
+
+            // this is a result with rows. Print the header and rows, watch for concurrent cancellation
+            printMetadataInfo(rowMetadata, colWidths, alignments, out);
+
+            int rowCount = 0;
+            for (SqlRow row : sqlResult) {
+                rowCount++;
+                printRow(row, colWidths, alignments, out);
+            }
+
+            // bottom line after all the rows
+            printSeparatorLine(sqlResult.getRowMetadata().getColumnCount(), colWidths, out);
+
+            String message = new AttributedStringBuilder()
+                    .style(AttributedStyle.BOLD.foreground(PRIMARY_COLOR))
+                    .append(String.valueOf(rowCount))
+                    .append(" row(s) selected")
+                    .toAnsi();
+            out.println(message);
+        } catch (HazelcastSqlException e) {
+            // the query failed to execute
+            String errorPrompt = new AttributedStringBuilder()
+                    .style(AttributedStyle.BOLD.foreground(PRIMARY_COLOR))
+                    .append(e.getMessage())
+                    .toAnsi();
+            out.println(errorPrompt);
+        }
+    }
+
+    private static int[] determineColumnWidths(SqlRowMetadata metadata) {
```

Comment:
Method seems incomplete. For example `DECIMAL_FORMAT_LENGTH` is unused, we use `TIMESTAMP_WITH_TIME_ZONE_FORMAT_LENGTH` for `DECIMAL`. Also at some types we ignore the header length. Maybe we could create `determineColumnWidth` that would do `max(min(header.length(), MAX_HEADER_LENGTH), typeLength)`.